apps/openmw/mwmechanics/actors.cpp

```diff
@@ -1200,17 +1197,16 @@ namespace MWMechanics
         }
 
         // Otherwise check if any actor in AI processing range sees the target actor
-        std::vector<MWWorld::Ptr> actors;
+        std::vector<MWWorld::Ptr> neighbors;
         osg::Vec3f position (actor.getRefData().getPosition().asVec3());
-        getObjectsInRange(position, mActorsProcessingRange, actors);
-        for(std::vector<MWWorld::Ptr>::iterator it = actors.begin(); it != actors.end(); ++it)
+        getObjectsInRange(position, mActorsProcessingRange, neighbors);
+        for (const MWWorld::Ptr &neighbor : neighbors)
         {
-            if (*it == actor)
+            if (neighbor == actor)
                 continue;
 
-            bool result =
-                MWBase::Environment::get().getWorld()->getLOS(*it, actor) &&
-                MWBase::Environment::get().getMechanicsManager()->awarenessCheck(actor, *it);
+            bool result = MWBase::Environment::get().getWorld()->getLOS(neighbor, actor);
+            result &= MWBase::Environment::get().getMechanicsManager()->awarenessCheck(actor, neighbor);
```

Comment:
This doesn't short-circuit any more, and there isn't a `&&=` operator in C++, so it can't be made to. Do we care?

Reply:
But &= is used in the same manner at line 495.